FrostCord/FrostCord
# Add Role Creation and Management

# Role Creation + Management
Adds a tabbed nav menu in server settings which allows for the creation/management of roles if the user has appropriate perms

# API changes
- Admin/Owner perms have been changed to be a composite of all perms below them for ease of checks
- Additional role store
- Addition to the user perms store to get the user's highest role
- Functions to increment/decrement a role heirarchy value

# Testing
- Verify you can create roles in servers you own
- Verify you can delete roles in servers you own
- Verify the EVERYONE and OWNER roles are read only
- Verify creating a new server creates these appropriately and in the right order
- Verify you cannot edit a role that is at or above your current highest role (ask me for an acct and I can let u sign into that foreasier testing than manually configuring ur roles in the db)
